module/item/item.js

```diff
@@ -0,0 +1,626 @@
+    }
+
+    /**
+     * Process weapon data for visuals
+     */
+    _processWeaponVisuals(itemData) {
+        const data = itemData.data;
+
+        // Formulate the visual shown for the weapon's magazines on the character sheet, either showing the ammo and the extra mags, ammo and the extra weapons, or just simple ammo
+        if (data.hasAmmo) {
+            if (this.actor?.data && this.consumesAmmunition) {
+                const ammoItem = this.findAmmunition();
+                if (ammoItem?.data)
+                    data.visualAmmo = data.ammo.value.toFixed(0) + " /" + ammoItem.data.data.quantity.toFixed(0) + "mag";
+                else {
+                    data.visualAmmo = data.ammo.value.toFixed(0);
+                    console.error("Somehow, the ammo item was received but turned up null: " + ammoItem);
+                }
+            }
+            else if ((data.ammo.extraMags ?? -2) >= 0)
+                data.visualAmmo = data.ammo.value.toFixed(0) + " /" + data.ammo.extraMags.toFixed(0) + "mag";
+            else if ((data.ammo.extraMags ?? -2) === -1 && data.quantity > 1)
+                data.visualAmmo = data.ammo.value.toFixed(0) + " + " + (data.quantity - 1).toFixed(0) + "pcs";
+            else
+                data.visualAmmo = data.ammo.value.toFixed(0);
+        } else { data.visualAmmo = ""; }
+
+        // Visual to-hit bonus
+        data.visualToHit = data.toHit >= 0 ? "+" + data.toHit.toString() : data.toHit.toString();
+    }
+
+    /* -------------------------------------------- */
+    /* Getters                                      */
+    /* -------------------------------------------- */
+
+    /**
+     * Does the Item have an area of effect target
+     * @type {boolean}
+     */
+    get hasAreaTarget() {
+        const target = this.data.data.target;
+        return target && (target.type in CONFIG.SGRPG.areaTargetTypes);
+    }
+
+    /**
+     * Does the Item implement an attack roll as part of its usage
+     * @type {boolean}
+     */
+    get hasAttack() {
+        return typeof this.data.data.attackAbility === "string";
+    }
+
+    /**
+     * Whether the weapon has a separate ammo item it consumes
+     */
+    get consumesAmmunition() {
+        if (!this.data.data.ammo) {
+            return false;
+        }
+        if (this.data.data.ammo.target === CONFIG.SGRPG.actionReloadValue) {
+            return false;
+        }
+        return this.data.data.ammo.target && Number.isNumeric(this.data.data.ammo.max);
+    }
+
+    /* -------------------------------------------- */
+    /* Item Use Functions                           */
+    /* -------------------------------------------- */
+
+    async roll() {
+        if (this.type === "equip" && this.actor && this.data.data.isLightItem) {
+            this.actor.changeLightSource(this);
+        } else {
+            this.displayCard();
+        }
```

Comment:
Sounds strange this to be roll function. I would prefer to refactor it into probably `useItem` function or something like that